client/client.go

```diff
@@ -368,6 +368,11 @@ func (c *Client) SetDebug(w io.Writer) {
 	c.conn.SetDebug(w)
 }
 
+// Conn returns the client's imap.Conn.
+func (c *Client) Conn() *imap.Conn {
```

Comment:
Thanks for looking at this and for the feedback!

Adding a Timeout field to imap.Client seems like it would be elegant. I am wondering though what it would really mean. It seems like to mean something similar to net/http it would mean setting a deadline for every command issued. Which might make this change a lot more complex.

I'm leaning towards adding a SetDeadline or changing this function to return net.Conn.

Having a function to access the net.Conn might be safest interface wise. We wouldn't be going too far down a path where the interface suggests that is the best way to handle timeouts (which adding a SetDeadline seems to imply). Having it return net.Conn might be the simplest bet (other than just using client.New!).

I'll do that tomorrow unless you have any further thoughts.

Thanks again.